wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java

```diff
@@ -109,16 +107,22 @@ public ProjectManager(VirtualFileSystemProvider vfsProvider,
 
     @PostConstruct
     void initWatcher() throws IOException {
-        FileWatcherNotificationListener defaultListener = new FileWatcherNotificationListener(VirtualFileFilter.ACCEPT_ALL) {
-            @Override
-            public void onFileWatcherEvent(VirtualFile virtualFile, FileWatcherEventType eventType) {
-                LOG.debug("FS event detected: " + eventType + " " + virtualFile.getPath().toString() + " " + virtualFile.isFile());
-                eventService.publish(DtoFactory.newDto(VfsWatchEvent.class)
-                                               .withPath(virtualFile.getPath().toString())
-                                               .withFile(virtualFile.isFile())
-                                               .withType(eventType));
-            }
-        };
+        FileWatcherNotificationListener defaultListener =
+                new FileWatcherNotificationListener(file -> !(file.getPath().toString().contains(".codenvy")
```

Comment:
@dkuleshov @vparfonov  What do you think about moving that component out of ProjectManager class. It is not obvious for me why it is placed here. Git and other components also related to it, not only project manager.